### DISCUSSION TITLE
[ENC] parsed INFORMATIONAL_V1 request 23781043 [ HASH N(INVAL_ID) ]

### DISCUSSION THREAD

**Discussion by atomus1990** [strongswan/strongswan#1530] [General]
I have a VPN configuration, that works intermittent.
Sometimes, the tunnels gets established, other times it does not and it fails with the error in subject.

It uses IKEv1, main mode, and PSK. 
I found this older thread:
https://wiki.strongswan.org/issues/2416

"It is well known that IKEv1 main mode with PSK authentication just does not work in NAT situations! Please switch to IKEv2 if the peer supports it or use certificate-based authentication."

Phase 1 always gets established, but intermittency is present for phase 2.

Is this a limitation or a workaround exist?

Regarding phase 2, I tried both exact subnets (local and remote) and 0.0.0.0/0

Another thing I would like to add. I have to identifical VPN configuration, each with 5 Phases 2 (one for each prefix). The VPN that uses remote identifier - the public IP works. The other one, that use remote identifier private IP has the errors mentioned

**Comment by tobiasbrunner** [comment]
> Sometimes, the tunnels gets established, other times it does not and it fails with the error in subject.

That's an error returned by the other peer, so check the logs there.

> Regarding phase 2, I tried both exact subnets (local and remote) and 0.0.0.0/0

You should definitely use the exact subnets. IKEv1 does not support narrowing.



> The VPN that uses remote identifier - the public IP works. The other one, that use remote identifier private IP has the errors mentioned

What do you mean with "remote identifier"?

Please provide more information (configs, logs etc., see #196).

**Comment by atomus1990** [comment]
By remote identifier, I mean:

      local {
         auth = psk
         id = X.X.X.X
      }
      remote {  <<<
         auth = psk
         id = X.X.X.X   
      }
 Regarding logs, I will reply a bit later, as I enabled a more verbose logging.

**Reply by tobiasbrunner** [reply]
OK, these identifiers are only relevant during phase 1. So if that's successful, we can probably ignore them.

**Comment by atomus1990** [comment]
Hi,

We eventually solved it. There was a routing issue on the other end, with the prefixes for VPN. Thx for your answer
